packages/light-react/src/stores/sendStore.js

```diff
@@ -0,0 +1,128 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+//
+// SPDX-License-Identifier: MIT
+
+import { action, computed, observable } from 'mobx';
+import { blockNumber$, post$ } from '@parity/light.js';
+import { isAddress } from '@parity/api/lib/util/address';
+import noop from 'lodash/noop';
+
+import parityStore from './parityStore';
+import tokensStore from './tokensStore';
+
+const DEFAULT_GAS = 21000; // Default gas amount to show
+
+class SendStore {
+  @observable blockNumber; // Current block number, used to calculate confirmations.
+  @observable estimated = DEFAULT_GAS; // Estimated gas amount for this transaction.
+  tx; // The actual tx we are sending. No need to be observable.
+  @observable txStatus; // Status of the tx, see wiki for details.
+
+  constructor () {
+    this.api = parityStore.api;
+  }
+
+  acceptRequest = password => {
+    // Avoid calling this method from a random place
+    if (!this.requestId) {
+      return Promise.reject(
+        new Error('The requestId has not been generated yet.')
+      );
+    }
+
+    // Since we accepted this request, we also start to listen to blockNumber,
+    // to calculate the number of confirmations
+    blockNumber$().subscribe(this.setBlockNumber);
+
+    return this.api.signer.confirmRequest(this.requestId, null, password);
+  };
+
+  /**
+   * Get the number of confirmations our transaction has.
+   */
+  @computed
+  get confirmations () {
+    if (!this.txStatus.confirmed) {
+      return -1;
+    }
+    return this.blockNumber - +this.txStatus.confirmed.blockNumber;
+  }
+
+  /**
+   * Estimate the amount of gas for a particular transaction. Note: at the time
+   * we're estimating, this.tx has not been set yet (only set at postTx stage),
+   * that's why we're not using this.tx here, but a tx passed as argument.
+   */
+  estimateGas = tx => {
+    // We only estimate txs to a correct address
+    if (!tx || !isAddress(tx.to)) {
+      return Promise.resolve(DEFAULT_GAS);
```

Comment:
It seems this would not estimate in the cases where the recipient could be a contract? (Which is a valid transfer, and probably >21k)